src/isl/isl_type_reference.rs

```diff
@@ -42,6 +63,21 @@ impl IslTypeRef {
         IslTypeRef::Anonymous(IslTypeImpl::new(None, constraints.into()))
     }
 
+    pub fn core_isl_type_from_type_name(type_name: &str) -> IslTypeRef {
+        // get the core types map and related text value for given type_name
+        let core_types_map: HashMap<&str, &str> = include!("core_types.rs");
+        let text = core_types_map.get(type_name).unwrap().to_owned();
```

Comment:
This will panic if the `type_name` is unexpected. If that's the intent, let's document it and use `expect` to provide context if it ever happens.

Reply:
This issues is resolved as preloading the `type_store` with all builtin types which doesn't use any `HashMap`